src/Orleans.Transactions/Abstractions/ITransactionalStateStorage.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Orleans.Concurrency;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Orleans.Transactions.Abstractions
+{
+    public interface ITransactionalStateStorage<TState>
+        where TState : class, new()
+    {
+        Task<TransactionalStorageLoadResponse<TState>> Load(string stateName);
+
+        Task<string> Persist(
+            string stateName,
+            string expectedETag,
+            string metadata,
+            List<PendingTransactionState<TState>> statesToPrepare);
+
+        Task<string> Confirm(
```

Comment:
Would the traditional 2PC names of Prepare/Commit be more suitable here?

Reply:
We intentionally deviated from the 2pc naming when dealing with the storage to make it easier to talk about the algorithm.  If we use the term 'prepare' to refer to a signal sent to the transactional resource and for how that resource writes to storage, nuance is lost or hard to communicate.  Now one may say on prepare the resource persists the change.

Reply:
Ok, sounds fine to me